authors/apps/authentication/views.py

```diff
@@ -44,7 +53,24 @@ def post(self, request):
         serializer = self.serializer_class(data=user)
         serializer.is_valid(raise_exception=True)
 
-        return Response(serializer.data, status=status.HTTP_200_OK)
+        token = generate_token(serializer.data).decode()
+        output = serializer.data
+        output['token'] = token
+        return Response(output, status=status.HTTP_200_OK)
+
+
+def generate_token(identity: dict):
+    """
+    Method that generates a JSON Web Token for the user
+    :param identity: User information to be encoded as a dictionary
+    :return: JWT token
+    :rtype: bytes
+    """
+    payload = dict(identity=identity,
```

Comment:
Instead of passing all user's information into the payload. Try passing email or user_id instead

Reply:
We pass in the email and username only. That is what is in the identity.